### DISCUSSION TITLE
Idea: Support Vulkan / SYCL / OpenCL for inference improving coverage for other GPUs e.g. intel, amd.

### DISCUSSION THREAD

**Discussion by ghchris2021** [EricLBuehler/mistral.rs#541] [Ideas]
Idea: Support Vulkan / SYCL / OpenCL for inference improving coverage for other GPUs e.g. intel, amd.

Intel discrete gpus are reaching their 2nd generation and their IGPUs similarly are evolving through their
2nd generation of using similar architectures now.  They support SYCL, OpenCL, vulkan as do other non-intel GPUs.

SYCL, Vulkan, and OpenCL are relatively vendor neutral accelerated compute platforms which can support many or all of the current GPU vendors (intel, amd, nvidia) as well as CPU based compute runtimes flexibly.

**Comment by sigma001** [comment]
i would heavily prefer vulkan for convenience..
